doc/source/serve/production.md

```diff
@@ -567,3 +557,216 @@ This section summarizes the best practices when deploying to production:
+    appStatus: {}
+    dashboardStatus:
+      healthLastUpdateTime: "2022-07-18T21:54:53Z"
+      lastUpdateTime: "2022-07-18T21:54:54Z"
+    rayClusterName: rayservice-sample-raycluster-bshfr
+    rayClusterStatus: {}
+...
+```
+
+In the status, you can see that the `RayService` is preparing a pending cluster.
+After the pending cluster is healthy, it will be switched to the active cluster and the previous cluster will be terminated.
+
+## Failure Recovery
+
+Ray Serve is resilient to any component failures within the Ray cluster out of the box.
+You can read more about how process and worker node failures are handled in: {ref}`serve-ft-detail`.
+
+### Head node failures
+
+By default the Ray head node is a single point of failure: if it crashes, the entire cluster crashes and needs to be restarted.
+When running on Kubernetes, the `RayService` controller will health check the cluster and restart it if this occurs, but it will still result in some downtime.
+In Ray 2.0, `KubeRay` has added experimental support for [GCS fault tolerance](https://ray-project.github.io/kuberay/guidance/gcs-ha/#ray-gcs-ha-experimental), preventing the Ray cluster from crashing if the head node goes down.
+While the head node is recovering, Serve applications can still handle traffic but will not be able to perform any updates or recover from other failures (e.g., actors or worker nodes crashing).
+Once the GCS is recovered, the cluster should return to normal behavior.
+
+### Health checking
+
+By default, each actor making up a Serve deployment is health checked and restarted on failure.
+
+:::{note}
+User-defined health checks are experimental and may be subject to change before the interface is stabilized. If you have any feedback or run into any issues or unexpected behaviors, please file an issue on GitHub.
+:::
+
+You can customize this behavior to perform an application-level health check or to adjust the frequency/timeout.
+To define a custom healthcheck, define a `check_health` method on your deployment class.
+This method should take no arguments and return no result, raising an exception if the replica should be considered unhealthy.
+You can also customize how frequently the health check is run and the timeout when a replica will be deemed unhealthy if it hasn't responded in the deployment options.
+
+> ```python
+> @serve.deployment(health_check_period_s=10, health_check_timeout_s=30)
+> class MyDeployment:
+>     def __init__(self, db_addr: str):
+>         self._my_db_connection = connect_to_db(db_addr)
+>
+>     def __call__(self, request):
+>         return self._do_something_cool()
+>
+>     # Will be called by Serve to check the health of the replica.
+>     def check_health(self):
+>         if not self._my_db_connection.is_connected():
+>             # The specific type of exception is not important.
+>             raise RuntimeError("uh-oh, DB connection is broken.")
+> ```
+
+[KubeRay]: https://ray-project.github.io/kuberay/
```

Comment:
Could we add a header here? These links are floating right now and not very related to the previous section:

<img width="771" alt="Screen Shot 2022-08-10 at 1 39 05 PM" src="https://user-images.githubusercontent.com/92341594/184015672-dac993bf-25d8-4a4f-b26b-1cab441d4810.png">

Reply:
yeah i'll fix these...